src/app/modules/main/wallet_section/module.nim

```diff
@@ -105,45 +110,59 @@ method delete*(self: Module) =
   self.sendModule.delete
   self.controller.delete
   self.view.delete
+
   if not self.addAccountModule.isNil:
     self.addAccountModule.delete
 
 method updateCurrency*(self: Module, currency: string) =
   self.controller.updateCurrency(currency)
 
-method switchAccount*(self: Module, accountIndex: int) =
-  self.assetsModule.switchAccount(accountIndex)
-  self.collectiblesModule.switchAccount(accountIndex)
-  self.transactionsModule.switchAccount(accountIndex)
-  self.overviewModule.switchAccount(accountIndex)
-
-method switchAccountByAddress*(self: Module, address: string) =
-  let accountIndex = self.controller.getIndex(address)
-  self.switchAccount(accountIndex)
-
 method setTotalCurrencyBalance*(self: Module) =
   self.view.setTotalCurrencyBalance(self.controller.getCurrencyBalance())
 
+method notifyFilterChanged(self: Module) =
+  self.overviewModule.filterChanged(self.filter.addresses, self.filter.chainIds)
+  self.assetsModule.filterChanged(self.filter.addresses, self.filter.chainIds)
+  self.collectiblesModule.filterChanged(self.filter.addresses, self.filter.chainIds)
+  self.transactionsModule.filterChanged(self.filter.addresses, self.filter.chainIds)
+
 method getCurrencyAmount*(self: Module, amount: float64, symbol: string): CurrencyAmount =
   return self.controller.getCurrencyAmount(amount, symbol)
 
+
+method setFilterAddress*(self: Module, address: string) =
+  self.filter.setAddress(address)
+  self.notifyFilterChanged()
+
 method load*(self: Module) =
   singletonInstance.engine.setRootContextProperty("walletSection", newQVariant(self.view))
 
+  self.events.on(SIGNAL_WALLET_ACCOUNT_UPDATED) do(e:Args):
+    self.notifyFilterChanged()
   self.events.on(SIGNAL_WALLET_ACCOUNT_SAVED) do(e:Args):
+    let args = AccountSaved(e)
     self.setTotalCurrencyBalance()
+    self.filter.setAddress(args.account.address)
+    self.notifyFilterChanged()
   self.events.on(SIGNAL_WALLET_ACCOUNT_DELETED) do(e:Args):
-    self.switchAccount(0)
+    let args = AccountDeleted(e)
     self.setTotalCurrencyBalance()
+    self.filter.removeAddress(args.address)
+    self.notifyFilterChanged()
   self.events.on(SIGNAL_WALLET_ACCOUNT_CURRENCY_UPDATED) do(e:Args):
     self.view.setCurrentCurrency(self.controller.getCurrency())
     self.setTotalCurrencyBalance()
+    self.notifyFilterChanged()
   self.events.on(SIGNAL_WALLET_ACCOUNT_NETWORK_ENABLED_UPDATED) do(e:Args):
+    self.filter.updateNetworks()
     self.setTotalCurrencyBalance()
+    self.notifyFilterChanged()
   self.events.on(SIGNAL_WALLET_ACCOUNT_TOKENS_REBUILT) do(e:Args):
     self.setTotalCurrencyBalance()
+    self.notifyFilterChanged()
```

Comment:
why this one?

Reply:
i could create an alias function, but the result is the same: reload current :)

Reply:
it would be clearer for anybody reading this code IMO